File: Data Structures & Algorithms/binary-search/submission-1.go
func search(nums []int, target int) int {
	l,r := 0, len(nums) - 1
	
	for r - l > 1 {
		mid := (l+r)/2
		if nums[mid] > target {
			r = mid
		} else if nums[mid] < target {
			l = mid
		} else if nums[mid] == target {
			return mid
		}
	}
	if nums[l] == target {
		return l
	} else if nums[r] == target {
		return r
	} else {
		return -1
	}
}

File: Data Structures & Algorithms/combination-target-sum/submission-1.py
from collections import Counter

class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        nums.sort()
        output = []

        def dfs(i: int, data: List[int], total: int) -> None:
            if total == target:
                output.append(data.copy())
                return
            
            for num in range(i, len(nums)):
                if total + nums[num] > target:
                    return
                data.append(nums[num])
                dfs(num, data, total + nums[num])
                data.pop()
        dfs(0, [], 0)

        return output